src/ngraph/op/fused/group_conv.cpp

```diff
@@ -49,10 +51,38 @@ op::GroupConvolution::GroupConvolution(const Output<Node>& data_batch,
     constructor_validate_and_infer_types();
 }
 
+op::GroupConvolution::GroupConvolution(const Output<Node>& data_batch,
+                                       const Output<Node>& filters,
+                                       const Strides& window_movement_strides,
+                                       const Strides& window_dilation_strides,
+                                       const CoordinateDiff& padding_below,
+                                       const CoordinateDiff& padding_above,
+                                       const Strides& data_dilation_strides,
+                                       const PadType& pad_type)
+    : FusedOp({data_batch, filters})
+    , m_window_movement_strides(window_movement_strides)
+    , m_window_dilation_strides(window_dilation_strides)
+    , m_padding_below(padding_below)
+    , m_padding_above(padding_above)
+    , m_data_dilation_strides(data_dilation_strides)
+    , m_groups(NOT_INITIALIZED_GROUPS)
+    , m_pad_type(pad_type)
+{
+    constructor_validate_and_infer_types();
+}
+
 void op::GroupConvolution::pre_validate_and_infer_types()
 {
     auto data_shape = get_input_partial_shape(0);
     auto filters_shape = get_input_partial_shape(1);
+
+    if (has_groups_in_filters_shape())
+    {
+        NODE_VALIDATION_CHECK(
+            this, filters_shape[0].is_static(), "Groups stored in filters_shape must be static");
```

Comment:
We need to find a way to support this case. It would be better to allow validation to pass, but throw an exception in `get_groups` if there is not enough information. We could probably do this by internally storing `m_groups` as a `Dimension` type, and throwing an exception if that `Dimension` is dynamic.

Reply:
I modified `m_groups` to be internally stored as a dimension.